kotlin/reflect/src/main/java/com/squareup/moshi/kotlin/reflect/KotlinJsonAdapter.kt

```diff
@@ -56,11 +56,14 @@ private val ABSENT_VALUE = Any()
 internal class KotlinJsonAdapter<T>(
     val constructor: KFunction<T>,
     val bindings: List<Binding<T, Any?>?>,
-    val options: JsonReader.Options) : JsonAdapter<T>() {
+    val options: JsonReader.Options,
+    val failureStrategy: FailureStrategy) : JsonAdapter<T>() {
 
   override fun fromJson(reader: JsonReader): T {
     val constructorSize = constructor.parameters.size
 
+    // accumulator for Non-null values that are null or required fields that are missing
+    val failures = if(failureStrategy == FailureStrategy.ACCUMULATE_ERRORS) mutableListOf<String>() else null
```

Comment:
This should _always_ be initialized to `null` and lazily instantiated only if an error occurs.